### ISSUE TITLE
How to train on my own dataset?

### ISSUE THREAD

**Issue by zp19990818** [suhwan-cho/TMO#2]
Hi, 

Your work is really great. But how to train on my own dataset? My dataset likes DUTS， which masks are binary. I noticed that it is the validation set because it is only validated on DAVIS. Can you tell me where I can modify the color palette for semantic segmentation?

**Comment by zp19990818** [comment]
If I don't change the label palette, the console output is mostly 0 for loss and iou
![image](https://github.com/suhwan-cho/TMO/assets/53686038/c69ed959-0311-45f0-b5ba-f295ca7fdee2)

**Comment by zp19990818** [comment]
Or maybe there is an error somewhere else? I would appreciate any help you can provide!

**Comment by suhwan-cho** [comment]
Hi, sorry for the late reply.

TMO uses DAVIS and DUTS during network training. As you said, they have different formats so we have to deal with them differently. As can be seen at the data-loading files, DAVIS masks are generated based on "palette" manner while DUTS masks are generated based on a simple quantization manner, i.e., 0 or 1. 

If you want to use semantic segmentation datasets during training, you should select a single target class and convert the ground truth masks to binary format (1 for the target class and 0 for others).

**Comment by zp19990818** [comment]
> Hi, sorry for the late reply.
> 
> TMO uses DAVIS and DUTS during network training. As you said, they have different formats so we have to deal with them differently. As can be seen at the data-loading files, DAVIS masks are generated based on "palette" manner while DUTS masks are generated based on a simple quantization manner, i.e., 0 or 1.
> 
> If you want to use semantic segmentation datasets during training, you should select a single target class and convert the ground truth masks to binary format (1 for the target class and 0 for others).

Thank you very much! I'll look into this section more closely.
